## **Exercice 1 : Analyse de données de base avec Kaggle**

In [1]:
import pandas as pd

In [2]:
df_boston = pd.read_csv('housing.csv')

In [3]:
df_boston.head()

,RM,LSTAT,PTRATIO,MEDV
0,6.575,4.98,15.3,504000.0
1,6.421,9.14,17.8,453600.0
2,7.185,4.03,17.8,728700.0
3,6.998,2.94,18.7,701400.0
4,7.147,5.33,18.7,760200.0


In [4]:
df_boston['RM'].describe()

count    489.000000
mean       6.240288
std        0.643650
min        3.561000
25%        5.880000
50%        6.185000
75%        6.575000
max        8.398000
Name: RM, dtype: float64

On obtient une moyenne de  6.240288 pour 489 observations, l'écart type est  3.561 et la médiane est 6.185

## **Exercice 2 : Analyse comparative des données de vente au détail**

In [5]:
df_feature = pd.read_csv('Retail/Features data set.csv')

In [6]:
df_feature.head(5)

,Store,Date,Temperature,Fuel_Price,MarkDown1,MarkDown2,MarkDown3,MarkDown4,MarkDown5,CPI,Unemployment,IsHoliday
0,1,05/02/2010,42.31,2.572,NaN,NaN,NaN,NaN,NaN,211.096358,8.106,False
1,1,12/02/2010,38.51,2.548,NaN,NaN,NaN,NaN,NaN,211.242170,8.106,True
2,1,19/02/2010,39.93,2.514,NaN,NaN,NaN,NaN,NaN,211.289143,8.106,False
3,1,26/02/2010,46.63,2.561,NaN,NaN,NaN,NaN,NaN,211.319643,8.106,False
4,1,05/03/2010,46.50,2.625,NaN,NaN,NaN,NaN,NaN,211.350143,8.106,False


In [7]:
df_feature.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8190 entries, 0 to 8189
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Store         8190 non-null   int64  
 1   Date          8190 non-null   object 
 2   Temperature   8190 non-null   float64
 3   Fuel_Price    8190 non-null   float64
 4   MarkDown1     4032 non-null   float64
 5   MarkDown2     2921 non-null   float64
 6   MarkDown3     3613 non-null   float64
 7   MarkDown4     3464 non-null   float64
 8   MarkDown5     4050 non-null   float64
 9   CPI           7605 non-null   float64
 10  Unemployment  7605 non-null   float64
 11  IsHoliday     8190 non-null   bool   
dtypes: bool(1), float64(9), int64(1), object(1)
memory usage: 712.0+ KB


In [8]:
df_feature['IsHoliday'].unique()

array([False,  True])

Dans le retail, le fichier feature comporte 8190 échantillons, de plus dans ce fichier, il y 12 variables qui peuvent être répartis en variables qualitative (Date, IsHoliday) et variables quantitative (Store, température, CPI, etc.).

In [9]:
df_feature['IsHoliday'].value_counts()

IsHoliday
False    7605
True      585
Name: count, dtype: int64

In [10]:
df_feature.describe().T

,count,mean,std,min,25%,50%,75%,max
Store,8190.0,23.000000,12.987966,1.000,12.000000,23.000000,34.000000,45.000000
Temperature,8190.0,59.356198,18.678607,-7.290,45.902500,60.710000,73.880000,101.950000
Fuel_Price,8190.0,3.405992,0.431337,2.472,3.041000,3.513000,3.743000,4.468000
MarkDown1,4032.0,7032.371786,9262.747448,-2781.450,1577.532500,4743.580000,8923.310000,103184.980000
MarkDown2,2921.0,3384.176594,8793.583016,-265.760,68.880000,364.570000,2153.350000,104519.540000
MarkDown3,3613.0,1760.100180,11276.462208,-179.260,6.600000,36.260000,163.150000,149483.310000
MarkDown4,3464.0,3292.935886,6792.329861,0.220,304.687500,1176.425000,3310.007500,67474.850000
MarkDown5,4050.0,4132.216422,13086.690278,-185.170,1440.827500,2727.135000,4832.555000,771448.100000
CPI,7605.0,172.460809,39.738346,126.064,132.364839,182.764003,213.932412,228.976456
Unemployment,7605.0,7.826821,1.877259,3.684,6.634000,7.806000,8.567000,14.313000


La moyenne de vente par magasin est de 23

In [17]:
df_ecommerce = pd.read_csv('ecommerce.csv')

In [18]:
df_ecommerce.head()

,Unnamed: 0,Clothing ID,Age,Title,Review Text,Rating,Recommended IND,Positive Feedback Count,Division Name,Department Name,Class Name
0,0,767,33,NaN,Absolutely wonderful - silky and sexy and comf...,4,1,0,Initmates,Intimate,Intimates
1,1,1080,34,NaN,Love this dress! it's sooo pretty. i happene...,5,1,4,General,Dresses,Dresses
2,2,1077,60,Some major design flaws,I had such high hopes for this dress and reall...,3,0,0,General,Dresses,Dresses
3,3,1049,50,My favorite buy!,"I love, love, love this jumpsuit. it's fun, fl...",5,1,0,General Petite,Bottoms,Pants
4,4,847,47,Flattering shirt,This shirt is very flattering to all due to th...,5,1,6,General,Tops,Blouses


In [21]:
df_ecommerce['Title'].value_counts

<bound method IndexOpsMixin.value_counts of 0                                                      NaN
1                                                      NaN
2                                  Some major design flaws
3                                         My favorite buy!
4                                         Flattering shirt
                               ...                        
23481                       Great dress for many occasions
23482                           Wish it was made of cotton
23483                                Cute, but see through
23484    Very cute dress, perfect for summer parties an...
23485                      Please make more like this one!
Name: Title, Length: 23486, dtype: object>

## **Exerice 3**

In [ ]:
df = pd.read_csv('data.csv',encoding="latin1")

In [ ]:
df.head()

In [ ]:
df.info()

Pour les données structuerer, on peut avoir la date, Quantity, CustomerID, Country

Pour compléter on peut ajouer des données sémi structuré comme les logs ou ajouter des données non strutucturé répesentant les produits du magasin

Ces données pourrait être utilile si, par exemple, on a un ensemple de produit et on veut pas parcourir à chaque fois un pour voir la photo du produit

## **Exerice 4**

In [22]:
df_rating= pd.read_csv('data/rating.csv')

In [23]:
df_rating.head()

,userId,movieId,rating,timestamp
0,1,2,3.5,2005-04-02 23:53:47
1,1,29,3.5,2005-04-02 23:31:16
2,1,32,3.5,2005-04-02 23:33:39
3,1,47,3.5,2005-04-02 23:32:07
4,1,50,3.5,2005-04-02 23:29:40


Les éléments structurés, on peut avoir le timestamp, le rating, le userId et le movieID

Lorsqu'on prend par exemple le timestant on remarque que les donnnées pas en vrague dans le csv mais sont bien structuré en fonction du temps, on peut aussi avoir le rating

## **Exerice 6**

In [ ]:
!pip install faker

In [ ]:
import numpy as np
from faker import Faker
import pandas as pd
import random
import uuid

In [ ]:
fake = Faker('fr_FR')
unique_id = uuid.uuid4()

# Generating synthetic names and ages
data = [(unique_id, fake.name(), fake.paragraph(), np.random.randint(1, 500)) for _ in range(500)]

In [ ]:
# Converting to a DataFrame
df = pd.DataFrame(data, columns=['ID', 'Name','Paragraph', 'Price'])


In [ ]:
df.head()